File: Encapsulation and Polymorphism/01. Shapes/Circle.cs
namespace Shapes
{
    using System;

    class Circle : BasicShape
    {
        public Circle(double radius)
            :base(radius, radius)
        {
        }

        public override double CalculateArea()
        {
            double circleArea = Math.PI * Math.Pow(this.Width, 2);

            return Math.Round(circleArea, 2);
        }

        public override double CalculatePerimeter()
        {
            double circlePerimeter = 2 * Math.PI * Width;

            return Math.Round(circlePerimeter, 2);
        }
    }
}

File: OOP in Java/Geometry/src/geometry2d/Circle.java
package geometry2d;

public class Circle extends PlaneShape {
	private double radius;

	public Circle(Vertex2D center, double radius) throws Exception {
		super(center);
		this.setRadius(radius);
	}

	public double getRadius() {
		return radius;
	}

	public void setRadius(double radius) throws Exception {
		if (radius <= 0) {
			throw new Exception("Circle radius must be positive number!");
		}
		
		this.radius = radius;
	}

	public double calculatePerimeter() {
		return 2 * Math.PI * this.radius;
	}

	@Override
	public double calculateArea() {
		return Math.PI * Math.pow(this.radius, 2);
	}
	
	@Override
	public String toString() {
		return String.format("Circle(Center: %s), Radius: %.2f, Perimeter: %.2f, Area: %.2f",
				this.vertices.get(0), this.radius, this.calculatePerimeter(), this.calculateArea());
	}	
}
